src/pixelator/cli/logging.py

```diff
@@ -0,0 +1,273 @@
+            # We only enter this if we didn't 'break' above during the while loop!
+            self._listener_process.terminate()
+
+    def initialize(self):
+        """Configure logging and start the listener process."""
+        # We do not need the listener process if there is no file to log to
+        if self.log_file:
+            self._listener_process.start()
+
+        handlers = []
+        handlers.append(logging.handlers.QueueHandler(self._queue))
+        self._root_logger.setLevel(logging.DEBUG if self.verbose else logging.INFO)
+
+        console_handler = ClickHandler()
+        if self.verbose:
+            console_handler.setFormatter(ColorFormatter(datefmt="%Y-%m-%d %H:%M:%S"))
+        else:
+            console_handler.setFormatter(DefaultCliFormatter())
+
+        handlers.append(console_handler)
+        self._root_logger.handlers = handlers
+        atexit.register(self._shutdown_listener)
+
+    def __enter__(self):
+        """Enter the context manager.
+
+        This will initialize the logging setup.
+        """
+        self.initialize()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Exit the context manager.
+
+        This will shut down the logging process if needed.
+        """
+        self._shutdown_listener()
+        atexit.unregister(self._shutdown_listener)
+        # Reraise exception higher up the stack
+        return False
+
+    @staticmethod
+    def _listener_logging_setup(log_file: Path | None, verbose: bool):
+        """Initialize the logging in the listener process."""
+        logger = logging.getLogger()
+
+        if log_file:
+            handler = logging.FileHandler(str(log_file), mode="w")
+            formatter = logging.Formatter(
+                "%(asctime)s %(processName)-10s %(name)s %(levelname)-8s %(message)s"
+            )
+            handler.setFormatter(formatter)
+            logger.addHandler(handler)
+            logger.setLevel(logging.DEBUG if verbose else logging.INFO)
+
+    @staticmethod
+    def _listener_process_main(
+        queue: multiprocessing.Queue, configure: Callable[[], None]
+    ):
+        """Entrypoint for the logging process.
+
+        :param queue: the queue to read log messages from
+        :param configure: the function to call to configure the logging
+        """
+        configure()
+
+        logging.info("Logging process started.")
+        while True:
+            try:
+                record = queue.get()
+                if (
+                    record is None
+                ):  # We send this as a sentinel to tell the listener to quit.
+                    break
+                logger = logging.getLogger(record.name)
+                logger.handle(record)  # No level or filter logic applied - just do it!
+            except Exception:
+                import traceback
+
+                click_echo("An unexpected error occurred in the log handler")
+                click_echo(traceback.format_exc())
```

Comment:
True, i think plain print statements are needed here. We can assume that the logging is broken if we ever reach this point so using log messages wil probably throw the same error in the error handler again. 

Since this is run in a separate process the print statements will usually get lost unless we do something with the stdout/stderr from the process, but i still think it is usefull to have them there. The reraised excpetion should be able to get the traceback to the main process and printed/logged there when the logging process is joined.